AppController/djinn.rb

```diff
@@ -3679,14 +3610,19 @@ def start_stop_api_services
 
     if my_node.is_db_master? or my_node.is_db_slave?
       db_master = nil
-      @nodes.each { |node|
-        db_master = node.private_ip if node.jobs.include?('db_master')
+      @state_change_lock.synchronize {
+        @nodes.each { |node|
+          db_master = node.private_ip if node.jobs.include?('db_master')
+        }
       }
       setup_db_config_files(db_master)
 
       threads << Thread.new {
         Djinn.log_info("Starting database services.")
-        db_nodes = @nodes.count{|node| node.is_db_master? or node.is_db_slave?}
+        db_nodes = nil
```

Comment:
Ok, it is required, otherwise the db_nodes variable won't be accessible outside the synchronized scope.